lib/manager/composer/artifacts.ts

```diff
@@ -22,6 +20,42 @@ import { getRepoStatus } from '../../util/git';
 import * as hostRules from '../../util/host-rules';
 import { UpdateArtifact, UpdateArtifactsResult } from '../common';
 
+function getAuthJson(): string | null {
+  const authJson = {};
+
+  const githubCredentials = hostRules.find({
+    hostType: PLATFORM_TYPE_GITHUB,
+    url: 'https://api.github.com/',
+  });
+  if (githubCredentials?.token) {
+    authJson['github-oauth'] = {
+      'github.com': githubCredentials.token,
+    };
+  }
+
+  const gitlabCredentials = hostRules.find({
+    hostType: PLATFORM_TYPE_GITLAB,
+    url: 'https://gitlab.com/api/v4/',
+  });
+  if (gitlabCredentials?.token) {
+    authJson['gitlab-token'] = {
+      'gitlab.com': gitlabCredentials.token,
+    };
+  }
+
+  hostRules
+    .findAll({ hostType: datasourcePackagist.id })
```

Comment:
I tried to add hostRules for a git repo in `renovate.json`, but as we only consider credentials of `hostType: packagist` here it never arrived in composer. I wonder if it makes sense to loosen this check a little, by either removing the filter at all or also adding hostrules where `hostType == null`?

This is what I tried in `renovate.json`:
```
  "hostRules": [
    {
      "domainName": "bitbucketserver",
      "username": "composer",
      "password": "xxx"
    }
  ]
```

Reply:
@danez You need to add `hostType: packagist` to your package rule

Reply:
Okay